### PR TITLE
Update balance proof in messaging.rst

### LINKED ISSUE #57: Update balance proof definitions in the messaging spec
Scope: keep the spec DRY.

- balance proof definitions from https://github.com/raiden-network/spec/pull/54 should be moved in the `messaging` spec. 
- `terminology` spec should also be updated and it should reference the complete definition of the terms from `messaging`, while also providing a short definition.
- any other spec that links these terms should use a link to the `terminology` term.

### LINKED ISSUE #61: Update balance proof definition
- balance proof definition in the messaging spec is now outdated 
https://github.com/raiden-network/spec/blob/master/messaging.rst#balance-proof
- it must be replaced with the new one from the smart contracts spec https://github.com/raiden-network/spec/blob/master/smart_contracts.rst#data-types-definition
- the terminology definition should be revised and should include a link to the definition (messaging spec)
- all references to the balance proof data types should link to the terminology
- definition should be removed from the smart contracts spec

### PR DESCRIPTION
**Pull request by palango**
Fixes #57 
Fixes #61 

## TODO
- [x] Link to messages from the terminology item
- [x] Remove BP description from the SC page
- [x] Link the terminology item when talking about the BP everywhere

### CODE REVIEW

File: messaging.rst [hunk 1]
```diff
@@ -17,34 +17,48 @@ Data required by the smart contracts to update the payment channel end of the pa
...
-|  signature            | bytes       | An elliptic curve 256k1 signature                                             |
-+-----------------------+-------------+-------------------------------------------------------------------------------+
++------------------------+------------+--------------------------------------------------------------+
+| Field Name             | Field Type |  Description                                                 |
++========================+============+==============================================================+
+|  balance_hash          | bytes32    | Balance data hash                                            |
++------------------------+------------+--------------------------------------------------------------+
+|  nonce                 | uint64     | Strictly monotonic value used to order transfers             |
++------------------------+------------+--------------------------------------------------------------+
+|  additional_hash       | bytes32    | Computed from the message. Used for message authentication   |
```
Comment (loredanacirstea) [inline messaging.rst]: Definition should be kept as `Hash of additional data used on the application layer, e.g.: payment metadata`

File: messaging.rst [hunk 2]
```diff
@@ -17,34 +17,48 @@ Data required by the smart contracts to update the payment channel end of the pa
...
+|  signature             | bytes      | Elliptic Curve 256k1 signature on the above data             |
++------------------------+------------+--------------------------------------------------------------+
+
+Balance Data Hash
+^^^^^^^^^^^^^^^^^
+
+``balance_hash`` = ``keccak256(transferred_amount || locked_amount || locksroot)``
+
+- :term:`Transferred amount` starts at 0 and is monotonic.
+- :term:`Locksroot` is the root node of the merkle tree of current pending locks.
```
Comment (loredanacirstea) [inline messaging.rst]: Please also define `locked_amount` here

Comment (loredanacirstea) [inline messaging.rst]: Also defined below.

File: terminology.rst [hunk 3]
```diff
@@ -70,7 +70,7 @@ Raiden Terminology
    balance proof
    Participant Balance Proof
    BP
-       Signed data required by the Payment Channel to prove the balance of one of the parties.
+       Signed data required by the :term:`Payment Channel` to prove the balance of one of the parties.
```
Comment (loredanacirstea) [inline terminology.rst]: Needs to reference the balance proof definition from messaging.rst

**Comment by palango** [comment]
Issues are fixed.

**Comment by LefterisJP** [comment]
@palango Thank you.

**Pull request merged**